lib/src/stream_subscription_transformer.dart

```diff
@@ -18,10 +18,14 @@ typedef void _SimpleHandler<T>(StreamSubscription<T> inner);
 /// [StreamSubscription.resume] is called, the corresponding handler is invoked.
 /// By default, handlers just forward to the underlying subscription.
 ///
-/// This guarantees that once the transformed [StreamSubscription] has been
-/// canceled, no further callbacks will be invoked. The [handlePause] and
-/// [handleResume] are invoked regardless of whether the subscription is paused
-/// already or not.
+/// Guarantees that once the transformed [StreamSubscription] has been canceled,
+/// no further callbacks will be invoked. The [handlePause] and [handleResume]
+/// are invoked regardless of whether the subscription is paused already or not.
+///
+/// In order to preserve [StreamSubscription] guarantees, **all callbacks must
+/// synchronously call the corresponding method** on the inner
```

Comment:
If that is the requirement, why don't you just do it for them, and let the handlers just be notifications?

I was assuming that the handlers were needed because you might want to *not* call the corresponding method, but do something else (e.g., direct events somewhere else during pause or pause instead of cancelling, so the subscription can be reused somewhere else - but then you really need to break the connection better, so you can't call through to the inner subscription after a cancel at all).